src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -201,6 +215,21 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
                                       completion:(MTRDeviceOpenCommissioningWindowHandler)completion
     MTR_AVAILABLE(ios(17.0), macos(14.0), watchos(10.0), tvos(17.0));
 
+/**
+ * Requests diagnostic logs of the desired types from the device.
+ *
+ * @param types      The types of logs being requested.
+ * @param timeout    The timeout for getting the logs.  If the timeout expires, completion
+ *                   will be called with whatever logs have been retrieved by that point
+ *                   (which might be none, and might include partial logs).
+ * @param queue      The queue on which completion will be called.
+ * @param completion The completion that will be called to pass in the file paths for the requested logs.
+ */
+- (void)getDiagnosticLogsOfTypes:(MTRDiagnosticLogTypes)types
+                         timeout:(NSTimeInterval * _Nullable)timeout
```

Comment:
```suggestion
                         timeout:(NSTimeInterval)timeout
```
NSTimeInterval is not an NSObject type.

Unless we want to allow not supplying a timeout?  In which case we should make this a nullable NSNumber storing an NSTimeInterval, I would think.

Reply:
i think we would like to enforce a timeout. Ideally we could default to something. but makes sense for the requestor to say i need logs in x amount of time. but then there could be some clients that say give me the logs i don't care about a timeout.

Reply:
will hold off on this and revert to NSInteger _Nullable is needed.

Reply:
done. will revert if needed